### PR TITLE
Use Moodle 4.5 Core AI subsystem

### PR DESCRIPTION
**Pull request by PM84**
Conditionally use tool_aiconnect or moodle 4.5 coreai subsystem A new function called perform_request checks the usemebisai setting and if it is set then the mebis-lp ai subsystem is used, otherwise the Core AI subsystem is used. The tool_aiconnect approach has been removed

### CODE REVIEW

File: lang/en/qtype_aitext.php [hunk 1]
```diff
@@ -103,6 +103,8 @@
 $string['showprompt'] = 'Show prompt';
 $string['thedefaultmarksscheme'] = 'Deduct a point from the total score for each grammar or spelling mistake.';
 $string['thedefaultprompt'] = 'Explain if there is anything wrong with the grammar and spelling in the text.';
+$string['usemebis_ai'] = 'Use Mebis AI';
```
Comment (PM84) [inline lang/en/qtype_aitext.php:106]:
```suggestion
$string['usemebis_ai'] = 'Use local_ai_manager';
```

File: lang/en/qtype_aitext.php [hunk 2]
```diff
@@ -103,6 +103,8 @@
...
+$string['usemebis_ai_setting'] = 'Use the Local aiconnector plugin from Mebis-lp to process AI related queries (must be installed)';
```
Comment (PM84) [inline lang/en/qtype_aitext.php:107]:
```suggestion
$string['use_local_ai_manager_setting'] = 'Use the local_ai_manager plugin to process AI related queries and tenancy mode (must be installed)';
```

File: settings.php [hunk 3]
```diff
@@ -66,6 +66,12 @@
         new lang_string('responseformat_setting', 'qtype_aitext'),
         0, ['plain' => 'plain', 'editor' => 'editor', 'monospaced' => 'monospaced']
     ));
+    $settings->add(new admin_setting_configcheckbox(
+        'qtype_aitext/usemebisai',
+        new lang_string('usemebis_ai', 'qtype_aitext'),
+        new lang_string('usemebis_ai_setting', 'qtype_aitext'),
```
Comment (PM84) [inline settings.php:72]:
```suggestion
        new lang_string('use_local_ai_manager_setting', 'qtype_aitext'),
```

File: settings.php [hunk 4]
```diff
@@ -66,6 +66,12 @@
         new lang_string('responseformat_setting', 'qtype_aitext'),
         0, ['plain' => 'plain', 'editor' => 'editor', 'monospaced' => 'monospaced']
     ));
+    $settings->add(new admin_setting_configcheckbox(
+        'qtype_aitext/usemebisai',
+        new lang_string('usemebis_ai', 'qtype_aitext'),
```
Comment (PM84) [inline settings.php:71]:
```suggestion
        new lang_string('use_local_ai_manager', 'qtype_aitext'),
```

File: settings.php [hunk 5]
```diff
@@ -66,6 +66,12 @@
         new lang_string('responseformat_setting', 'qtype_aitext'),
         0, ['plain' => 'plain', 'editor' => 'editor', 'monospaced' => 'monospaced']
     ));
+    $settings->add(new admin_setting_configcheckbox(
+        'qtype_aitext/usemebisai',
```
Comment (PM84) [inline settings.php:70]:
```suggestion
        'qtype_aitext/uselocalaimanager',
```

File: question.php [hunk 6]
```diff
@@ -143,6 +143,41 @@ public function compute_final_grade($responses, $totaltries) {
     public function apply_attempt_state(question_attempt_step $step) {
         $this->step = $step;
     }
+    /**
+     * Call the llm using either the 4.5 core api
+     * or the the mebis ai depending on the usemebisai
```
Comment (PM84) [inline question.php:148]:
```suggestion
     * or the the mebis ai depending on the uselocalaimanager
```

File: question.php [hunk 7]
```diff
@@ -143,6 +143,41 @@ public function compute_final_grade($responses, $totaltries) {
...
+     * settings checkbox.
+     *
+     * @param string $prompt
+     * @return string $response
+     */
+    public function perform_request(string $prompt, string $purpose): string {
+        if (get_config('qtype_aitext', 'usemebisai')) {
```
Comment (PM84) [inline question.php:155]:
```suggestion
        if (get_config('qtype_aitext', 'uselocalaimanager')) {
```

**Review by PM84 — COMMENTED** [review]
Hi Marcus, 
we would like the local_ai_manager itself to be named there rather than “mebis”. 
Apart from that, I don't see any problems that would prevent us from merging our forums. 

However, we will make a few minor changes. We can then contribute them directly to you.

File: lang/en/qtype_aitext.php (hunk 1 above)
Comment (marcusgreen) [reply lang/en/qtype_aitext.php:106]: I like and approve

File: lang/en/qtype_aitext.php (hunk 2 above)
Comment (marcusgreen) [reply lang/en/qtype_aitext.php:107]: I like and approve

File: settings.php (hunk 3 above)
Comment (marcusgreen) [reply settings.php:72]: I like and approve

File: settings.php (hunk 4 above)
Comment (marcusgreen) [reply settings.php:71]: I like this as well, and approve

File: question.php (hunk 7 above)
Comment (marcusgreen) [reply question.php:155]: Yes

**Comment by marcusgreen** [comment]
This is excellent

**Comment by PhMemmel** [comment]
closed in favor of #6

**Pull request closed without merging**